lisa/sut_orchestrator/baremetal/platform_.py

```diff
@@ -0,0 +1,153 @@
+
+    def _prepare_environment(self, environment: Environment, log: Logger) -> bool:
+        return True
+
+    def _deploy_environment(self, environment: Environment, log: Logger) -> None:
+        # currently only support one cluster
+        assert self._baremetal_runbook.cluster, "no cluster is specified in the runbook"
+        cluster_instance = self._baremetal_runbook.cluster[0]
+
+        cluster = self.cluster_factory.create_by_runbook(cluster_instance)
+        assert cluster.runbook.client, "no client is specified in the runbook"
+
+        # copy build (shared, check if it's copied)
+        if self._baremetal_runbook.source and not self.source_path:
+            source = self.source_factory.create_by_runbook(
+                self._baremetal_runbook.source
+            )
+            self._log.debug(f"found build '{source.type_name()}', to expand runbook.")
+            self.source_path = source.download()
+
+        # ready checker cleanup
+        if cluster_instance.ready_checker:
+            ready_checker = self.ready_checker_factory.create_by_runbook(
+                cluster_instance.ready_checker
+            )
+            ready_checker.clean_up()
+
+        # copy build if source exists
+        if not self.copy_build:
+            cluster.copy(source_path=self.source_path)
+            self.copy_build = True
+
+        assert environment.runbook.nodes_requirement, "no node is specified"
+        for node_space in environment.runbook.nodes_requirement:
+            assert isinstance(
+                node_space, schema.NodeSpace
+            ), f"actual: {type(node_space)}"
+            environment.create_node_from_requirement(node_space)
+
+        index = 0
+        for node in environment.nodes.list():
+            node_context = get_node_context(node)
+            node_context.management_port = cluster.runbook.client[index].management_port
+            node_context.address = cluster.runbook.client[index].connection.address
+            node_context.port = cluster.runbook.client[index].connection.port
+            node_context.private_key_file = cluster.runbook.client[
+                index
+            ].connection.private_key_file
+            node_context.password = cluster.runbook.client[index].connection.password
+            node_context.username = cluster.runbook.client[index].connection.username
+            index = index + 1
+
+        # deploy cluster
+        cluster.deploy(environment)
+
+        if cluster_instance.ready_checker:
+            ready_checker = self.ready_checker_factory.create_by_runbook(
+                cluster_instance.ready_checker
+            )
+
+        index = 0
+        for node in environment.nodes.list():
+            # ready checker
+            if ready_checker:
+                ready_checker.is_ready(node)
+
+            # get ip address
+            if cluster_instance.ip_getter:
+                ip_getter_runbook = self.ip_getter_factory.create_by_runbook(
```

Comment:
The created object is not a `runbook`, so it should be called like `ip_getter`.

Reply:
updated